packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart

```diff
@@ -48,12 +50,23 @@ class AppSizeController {
   /// The node set as the analysis tab root.
   ///
   /// Used to build the treemap and the tree table for the analysis tab.
-  ValueListenable<TreemapNode?> get analysisRoot => _analysisRoot;
-  final _analysisRoot = ValueNotifier<TreemapNode?>(null);
+  final analysisRoot =
+      ValueNotifier<Selection<TreemapNode>>(Selection<TreemapNode>());
 
   void changeAnalysisRoot(TreemapNode? newRoot) {
-    _analysisRoot.value = newRoot;
-    if (newRoot == null) return;
+    if (newRoot == null) {
+      analysisRoot.value = Selection<TreemapNode>();
+      return;
+    }
+
+    final searchCondition = (TreemapNode n) => n == newRoot;
+    final int selectedRowIndex = newRoot.root.childCountToMatchingNode(
+      matchingNodeCondition: searchCondition,
+      includeCollapsedNodes: false,
+    );
+    final int newPos = selectedRowIndex * defaultRowHeight as int;
```

Comment:
this is calculating the pixel that the row starts at, and I think you just want the selected row index here index here. So this line is not needed, and below, you can just write 

`analysisRoot.value = Selection(node: newRoot, flatListIndex: selectedRowIndex);`

Reply:
also add `scrollIntoView: true` here when you are creating the instance of `Selection`

`Selection(node: newRoot, flatListIndex: selectedRowIndex, scrollIntoView: true)`